python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java

```diff
@@ -57,6 +60,63 @@ void test() {
     System.out.println("hello");
   }
 
+  @Test
+  void testTypeshedImports() {
+    var typeInferenceV2 = new TypeInferenceV2(new ProjectLevelTypeTable(ProjectLevelSymbolTable.empty()));
+
+    var context = TestPythonVisitorRunner.createContext(new File("src/test/resources/semantic/v2/typeshed_imports.py"));
+    var fileInput = context.rootTree();
+
+    var builder = new SymbolTableBuilderV2();
+    builder.visitFileInput(fileInput);
+    fileInput.accept(typeInferenceV2);
+
+    {
+      var statement = fileInput.statements().statements().get(0);
+      var importedNames = statement.children().get(1).children().get(0).children();
+      {
+        var item = importedNames.get(0);
+        Assertions.assertInstanceOf(Name.class, item);
+        var name = (Name) item;
+        var type = name.typeV2();
+        Assertions.assertInstanceOf(ModuleType.class, type);
```

Comment:
I guess we can also assert by name of the module here on top of the fact it's a `ModuleType`